src/ast/postprocessing/structured_documentation_reconstruction.ts

```diff
@@ -146,19 +196,47 @@ export class StructuredDocumentationReconstructingPostprocessor
             return;
         }
 
-        const structDocNode = this.reconstructor.process(node, source);
+        /**
+         * Skip reconstructing preceding strcutured documentation
+         * when related fields is already an instance of StructuredDocumentation.
+         */
+        if (!(node.documentation instanceof StructuredDocumentation)) {
+            const precedingGap = this.reconstructor.getPrecedingGapCoordinates(node);
+            const preceding = this.reconstructor.fragmentCoordsToStructDoc(precedingGap, source);
 
-        if (structDocNode === undefined) {
-            return;
+            if (preceding) {
+                preceding.id = context.lastId + 1;
+
+                context.register(preceding);
+
+                node.documentation = preceding;
+
+                if (preceding.parent !== node) {
+                    preceding.parent = node;
+                }
+            }
         }
 
-        structDocNode.id = context.lastId + 1;
+        /**
+         * Dangling structured documentation can only be located in statements,
+         * that may have nested children (`StatementWithChildren` or `ContractDefinition`).
+         */
+        if (node instanceof StatementWithChildren || node instanceof ContractDefinition) {
+            const danglingGap = this.reconstructor.getDanglingGapCoordinates(node);
+            const dangling = this.reconstructor.fragmentCoordsToStructDoc(danglingGap, source);
 
-        context.register(structDocNode);
+            if (dangling) {
+                dangling.id = context.lastId + 1;
 
-        node.documentation = structDocNode;
+                context.register(dangling);
 
-        structDocNode.parent = node;
+                node.danglingDocumentation = dangling;
+
+                if (dangling.parent !== node) {
```

Comment:
Same question about the if

Reply:
Will address

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/115/commits/872be86fefb3d356d2552929b0a260684ffd6801